File: OOP-Encapsulation_HristoChipev/Student.cs
using System;
namespace OOP_Encapsulation
{
    public class Student
    {
        private string name;
        private int age;
        private List<string> courses;

        public Student(string name, int age, string[] courses)
        {
            this.name = name;
            this.age = age;
            this.courses = new List<string>(courses);
        }

        public void AddCourse(string course)
        {
            courses.Add(course);
        }

        public void RemoveCourse(string course)
        {
            courses.Remove(course);
        }

        public string[] GetCourses()
        {
            return courses.ToArray();
        }

        public string Name
        {
            get { return name; }
        }

        public int Age
        {
            get { return age; }
        }
    }
}

File: OOP-first-lecure-RadoslavGeorgiev/src/MartiTaskLectureTwoOOPEncapsulation/Student.java
package MartiTaskLectureTwoOOPEncapsulation;

import java.util.ArrayList;

public class Student {
        private String name;
        private int age;
        private ArrayList<String> courses;

        public Student(String name, int age) {
            this.name = name;
            this.age = age;
            this.courses = new ArrayList<String>();
        }

        public void addCourse(String course) {
            this.courses.add(course);
        }

        public void removeCourse(String course) {
            this.courses.remove(course);
        }

        public String[] getCourses() {
            return this.courses.toArray(new String[0]);
        }

        public int getAge() {
            return this.age;
        }

        public String getName() {
            return this.name;
        }
    }

